spec/datadog/appsec/security_engine/runner_spec.rb

```diff
@@ -178,4 +178,16 @@
       end
     end
   end
+
+  describe '#finalize!' do
+    it 'releases the waf handle when an error occurs' do
+      # waf_context is lazily initialized, so we need to ensure it is called
+      runner.waf_context
+
+      expect(waf_context).to receive(:finalize!).and_raise(StandardError)
+      expect(thread_safe_ref).to receive(:release).with(waf_handle)
+
+      runner.finalize! rescue StandardError
```

Comment:
you can still test is as

```suggestion
expect { runner.finalize! }.to raise(StandardError)
```